libs/commons/src/main/kotlin/uk/gov/justice/digital/hmpps/retry/Retry.kt

```diff
@@ -1,13 +1,24 @@
 package uk.gov.justice.digital.hmpps.retry
 
-fun <T> retry(maxRetries: Int, code: () -> T): T {
-    var throwable: Throwable? = null
-    (1..maxRetries).forEach { _ ->
+import kotlin.reflect.KClass
+
+fun <T> retry(maxRetries: Int, exceptions: List<KClass<out Exception>> = listOf(Exception::class), code: () -> T): T {
+    var throwable: Throwable?
+    (1..maxRetries).forEach { count ->
         try {
             return code()
         } catch (e: Throwable) {
-            throwable = e
+            val matchedException = exceptions.firstOrNull { it.isInstance(e) }
+            throwable = if (matchedException != null && count < maxRetries) {
+                null
+            } else {
+                e
+            }
+
+            if (throwable != null) {
+                throw throwable!!
+            }
```

Comment:
I'm a bit confused by this... probably just me... where do we return the successful value outside the forEach?